### PR TITLE
Clear image update

### PR DESCRIPTION
**Pull request by kvark**

### CODE REVIEW

File: libportability-gfx/src/impls.rs [hunk 1]
```diff
@@ -3126,25 +3120,55 @@ pub extern "C" fn gfxCmdFillBuffer(
...
+        slice::from_raw_parts(pRanges, rangeCount as _)
+    };
+    commandBuffer.clear_image(
+        match *image {
+            Image::Image(ref image) => image,
+            Image::Unbound(_) => panic!("Bound image expected!"),
+        },
+        conv::map_image_layout(imageLayout),
+        unsafe { mem::transmute(*pColor) },
+        unsafe { mem::zeroed() },
```
Comment (grovesNL) [inline libportability-gfx/src/impls.rs]: Callers having to use `zeroed` here makes me wonder whether the API should simply accept `Option` (or we provide another way to make this cleaner)

File: libportability-gfx/src/conv.rs [hunk 2]
```diff
@@ -170,9 +170,17 @@ pub fn map_subresource_range(subresource: VkImageSubresourceRange) -> image::Sub
     image::SubresourceRange {
         aspects: map_aspect(subresource.aspectMask),
         levels: subresource.baseMipLevel as _
-            ..(subresource.baseMipLevel + subresource.levelCount) as _,
+            .. if subresource.levelCount as i32 == VK_REMAINING_MIP_LEVELS {
+                !0
```
Comment (grovesNL) [inline libportability-gfx/src/conv.rs]: Where do we handle the `!0` cases on the HAL side currently? For example, in the Metal backend we seem to iterate directly through this range without special handling for this case

File: libportability-gfx/src/impls.rs [hunk 3]
```diff
@@ -1756,8 +1749,9 @@ pub extern "C" fn gfxCreateGraphicsPipelines(
             depth_bounds: if dyn_states.iter().any(|&ds| ds == VkDynamicState::VK_DYNAMIC_STATE_DEPTH_BOUNDS) {
                 None
             } else {
-                None // TODO
-            }
+                unsafe { info.pDepthStencilState.as_ref() }
```
Comment (grovesNL) [inline libportability-gfx/src/impls.rs]: 👍 I had hoped it was something simple like that

File: libportability-gfx/src/conv.rs (hunk 2 above)
Comment (kvark) [reply libportability-gfx/src/conv.rs]: Metal does actually handle `!0` in `clear_image` (at least). It's not pretty, and I'm open to ideas on how to expose this better. One option would be to just always require the exact level/layer counts, and our portability layer can remember them.

File: libportability-gfx/src/impls.rs (hunk 1 above)
Comment (kvark) [reply libportability-gfx/src/impls.rs]: would probably be easier to implement `Default` on those

**Comment by kvark** [comment]
we'll continue polishing this. Thanks for the review!
bors r=grovesNL

also, enabling the metal capture feature is not needed. It should be ON by default in debug. I'll remove it next time I touch it.

**Pull request merged**